### PR TITLE
DM: add set servo pulse width

### PR DESCRIPTION
**Pull request by deanm1278**

### CODE REVIEW

File: servo.ts [hunk 1]
```diff
@@ -31,6 +31,26 @@ namespace crickit {
             dev.setPwmFreq(this._pin, 50);
             dev.analogWrite(this._pin, value);
         }
+
+        /**
+         * set the pulse width to the servo in microseconds
+         */
+        //% group="Servos"
+        //% weight=10
+        //% blockId=sawservosetpulse block="crickit set %servo pulse to (μs) %value"
```
Comment (pelikhan) [inline servo.ts]: you can move (us) after %value

File: servo.ts [hunk 2]
```diff
@@ -31,6 +31,26 @@ namespace crickit {
...
+        //% value.min=0 value.max=20000
+        //% servo.fieldEditor="gridpicker"
+        //% servo.fieldOptions.width=220
+        //% servo.fieldOptions.columns=2
+        setPulse(value: number) {
+            const dev = saw();
+            value = value | 0;
+            value = Math.clamp(0, 20000, value);
+            value = 3.2767 * value;
```
Comment (pelikhan) [inline servo.ts]: value = (3.2767 * value) | 0;

**Review by pelikhan — APPROVED** [review]

File: servo.ts (hunk 1 above)
Comment (deanm1278) [reply servo.ts]: ok will do (was matching the format to the corresponding block under the Pins -> Servo heading)

Comment (pelikhan) [reply servo.ts]: Right... we've improved our block format parser now and it can handle a proper ordering of words.

**Pull request merged**